model/Software/Classes/PackageVerification.md

```diff
@@ -0,0 +1,25 @@
+SPDX-License-Identifier: Community-Spec-1.0
+
+# PackageVerification
```

Comment:
Why do we need that layer? Can't we just use `IntegrityMethod` instead. This also preserves the flexibility of instead just using other existing VerificationMethods I think that this approach will limit flexibility.

I also think that this restriction goes in the wrong direction. This prevents the package from using other verification methods but aother Elements like SBOM is not prevented from using `PackageVerification`.

Reply:
If we would drop that layer in the class diagram, we would also not need to introduce `PackageHash` which is basically just a `Hash`

Reply:
> If we would drop that layer in the class diagram, we would also not need to introduce PackageHash which is basically just a Hash

I think we need `PackageHash` such that we can document the hash refers to the distributed document.

Reply:
> Why do we need that layer?

My thinking was we could have a restriction that the package requires an `IntegrityMethod` of type `PackageVerification`.  It also is a way of implementing a proxy for the enumeration types @jeff-schutt proposed.

I personally think it makes the spec easier to read by grouping the package integrity methods together, but I don't feel strongly about this.

Reply:
so it is semantically different to `/Core/Element/verifiedUsing`?


- `/Core/Element/verifiedUsing` is the method to verify the real world object that corrensponds to the Element
- `this` is used to hash the distributed document (you mean artifact?).

Reply:
> My thinking was we could have a restriction that the package requires an IntegrityMethod of type PackageVerification. 

wouldn't it be more important or helpful to say that non-Packages are not allowed to have an IntegrityMethod of type PackageVerification?

Reply:
(lets move one of the conversations to https://github.com/spdx/spdx-3-model/pull/498/files/8477afa0b39440a85059c39ad5d6e7dd1c718e95#r1332675795, to stop the interleaved threads)